techLogger.js

```diff
@@ -72,47 +74,29 @@ function _makeLogger(prefix) {
             return updated;
         },
 
-        debug: function() {
-            _log("debug", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        debug: (...args) => _log("debug", _prefixLog(args, logPrefix)),
 
-        info: function() {
-            _log("info", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        info: (...args) => _log("info", _prefixLog(args, logPrefix)),
 
-        notice: function() {
-            _log("notice", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        notice: (...args) => _log("notice", _prefixLog(args, logPrefix)),
 
-        warn: function() {
-            _log("warning", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        warn: (...args) => _log("warning", _prefixLog(args, logPrefix)),
 
-        error: function() {
-            _log("error", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        error: (...args) => _log("error", _prefixLog(args, logPrefix)),
 
-        crit: function() {
-            _log("crit", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        crit: (...args) => _log("crit", _prefixLog(args, logPrefix)),
 
-        alert: function() {
-            _log("alert", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
-        },
+        alert: (...args) => _log("alert", _prefixLog(args, logPrefix)),
 
-        emerg: function() {
-            _log("emerg", _prefixLog(Array.prototype.slice.call(arguments), logPrefix));
+        emerg(...args) {
+            return _log("emerg", _prefixLog(args, logPrefix));
         },
 
-        createExpressLoggerStream: function(level) {
-            return {
-                write: function(message /*, encoding */ ) {
-                    _log(level, message);
-                }
-            };
+        createExpressLoggerStream(level) {
+            return { write: (message /*, encoding */ ) => _log(level, message) };
         },
 
-        splash: function(app, configuration) {
+        splash(app, configuration) {
             splash(this, app, configuration);
         }
```

Comment:
While converting this `_makeLogger` function, that creates the `logger` object, to ES6 I got a context issue using arrow functions for the `splash`methods that is referencing `this`:

Initial code
```js
function _makeLogger(prefix) {
  return {  <---------------------------------,
   //...                                      |
      splash: function(app, configuration) {  |
   //            ,----------------------------'
        splash(this, app, configuration);
     }
  };
}
```
Using arrow function (to have less code)

```js
function _makeLogger(prefix) {  <---------------,
  return {                                      |
   //...                                        |
      splash: (app, configuration) =>  splash(this, app, configuration)
  };
}
```
After reading some docs I discovered a new method definition syntax

```js
function _makeLogger(prefix) {
  return {  <-----------------------,
   //...                            |
      splash(app, configuration) {  |
   //            ,------------------'
        splash(this, app, configuration);
     }
  };
}
```

In the end the logger we create could be composed of both arrow functions (when not referencing `this` and methods) and methods like this:

```js
function _makeLogger(prefix) {
  return { 
        //...
        debug: (...args) => _log("debug", _prefixLog(args, logPrefix)),

        info: (...args) => _log("info", _prefixLog(args, logPrefix)),

        notice: (...args) => _log("notice", _prefixLog(args, logPrefix)),

        warn: (...args) => _log("warning", _prefixLog(args, logPrefix)),

        error: (...args) => _log("error", _prefixLog(args, logPrefix)),

        crit: (...args) => _log("crit", _prefixLog(args, logPrefix)),

        alert: (...args) => _log("alert", _prefixLog(args, logPrefix)),

        // Just as an example
        emerg(...args) {
            return _log("emerg", _prefixLog(args, logPrefix));
        },

        createExpressLoggerStream(level) {
            return { write: (message /*, encoding */ ) => _log(level, message) };
        },

        splash(app, configuration) {
            splash(this, app, configuration);
        }
  };
}
```
So here's my question, does it bother you having a mix like that for object definition or do we allow mixing the syntax? 

Well... just before submitting this comment I re-read an article I found ([When 'not' to use arrow functions](https://rainsoft.io/when-not-to-use-arrow-functions-in-javascript/)) and they discourage using arrow functions for objects methods definitions :)

WDYT @jlescot @sgaestel @arenoux @KINFOO @fskandrani @bmiegemolle ?